sorc/copygb.fd/CMakeLists.txt

```diff
@@ -1,32 +1,16 @@
-
-if(IntelComp)
-  set(fortran_flags "-g" "-O3" "-r8" "-auto" "${OpenMP_Fortran_FLAGS}")
-elseif(GNUComp)
-  set(fortran_flags "-g" "-O3" "-fdefault-real-8" "${OpenMP_Fortran_FLAGS}")
-else()
-  message("unknown compiler!")
-  exit()
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "-g -r8 -auto")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
+  set(CMAKE_Fortran_FLAGS "-g -fdefault-real-8")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
 endif()
 
-set(exe_name copygb)
-
-file(GLOB fortran_src
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.f90
-  ${CMAKE_CURRENT_SOURCE_DIR}/*.F)
+set(fortran_src copygb.F)
 
+set(exe_name copygb)
 add_executable(${exe_name} ${fortran_src})
+target_link_libraries(${exe_name} bacio::bacio_4 w3nco::w3nco_4 ip::ip_4 sp::sp_4
+  ${OpenMP_Fortran_FLAGS})
```

Comment:
use `OpenMP::OpenMP_Fortran`
Also,  wrap this in an if block of  `OpenMP_Fortran_FOUND`